### DISCUSSION TITLE
Git status says "up to date with origin/master" after changes, but this is not the case

### DISCUSSION THREAD

**Discussion by bwinsto2** [community/community#23047] [New to GitHub]
<p>I recognize that this seems like a beginner issue, but posting here is my last resort: I’ve tried numerous solutions online, and nothing seems to work.</p>
<p>Clearly, something is off with branches. When I create changes to my repo on the browser, I can pull the changes in Mac terminal no problem.</p>
<p>However, when I make changes in the Terminal, adding/committing/pushing does not work. Git status gives me the message:</p>
<pre><code class="lang-auto">On branch master

Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
</code></pre>
<p>I’ve tried lots of methods online for resolving this, including: <code>git checkout HEAD</code>, <code>git fetch --all \  git reset --hard origin/master</code>, <code>git branch --set-upstream-to origin/master</code></p>
<p>None of these have resolved the problem for me. I would be very grateful for any help that someone could provide, and I will try to respond quickly in case you are wondering what the output of different commands are on my local config.</p>
<p>Thanks so much!</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="bwinsto2" data-post="1" data-topic="129228">
<div class="title">
<div class="quote-controls"></div>
 bwinsto2:</div>
<blockquote>
<p>However, when I make changes in the Terminal, adding/committing/pushing does not work.</p>
</blockquote>
</aside>
<p>Exactly how do they “not work”? Are you getting error messages? If so, please quote those, for each step, together with the exact command you are using.</p>
<aside class="quote no-group" data-username="bwinsto2" data-post="1" data-topic="129228">
<div class="title">
<div class="quote-controls"></div>
 bwinsto2:</div>
<blockquote>
<p>I’ve tried lots of methods online for resolving this, including: <code>git checkout HEAD</code> , <code>git fetch --all \  git reset --hard origin/master</code> , <code>git branch --set-upstream-to origin/master</code></p>
</blockquote>
</aside>
<p>Please be very careful trying random commands, at least one in that list is extremely destructive: <code>git reset --hard origin/master</code> sets your current branch to the state of <code>origin/master</code> and <em>discards all local changes</em>, including those already committed (!) to your current branch. You may be able to salvage such orphaned commits if you do it soon, before they are garbage collected.</p>

**Comment by bwinsto2** [comment]
<p>Thanks for the reply.</p>
<p>I just cloned the repo, and I don’t have any meaningful local changes.</p>
<p>When I say the changes don’t work, what I mean is when I make changes, I get the message I copied above (aka the changes are not committing/being added to working tree(?)</p>
<p>For example, I could do <code>mkdir hi</code> then git status will yield above message.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Ah, I see. First off, git won’t track empty directories, so that <code>hi/</code> directory not showing up is fine. What happens if you create a file, or modify one that’s in the repository? Does <code>git show</code> mention that? If not, also look for a <code>.gitignore</code> file, if there is one. If your file matches any of the patterns in there it will be ignored.</p>
<p>For terms, “working tree” refers to the files you can, well, <em>work</em> on. New or edited files (or parts of them) on the working tree may then be added to the “index”, and committed.</p>

**Reply by PedroHJB** [reply]
Thank you very much!

**Comment by bwinsto2** [comment]
<p>Wow, do I feel dumb…</p>
<p>I was creating empty directories to check whether git was tracking changes. Just tried creating a file and everything worked.</p>
<p>Thanks so much for the help!!</p>

**Comment by adesalegn** [comment]
<p>Hi all,</p>
<p>I would like to ask the same question about a different problem.<br>
By mistake, I deleted my local git repository. Then, I used the <code>git clone</code> command to clone it from my remote repository. After cloning, I moved a few files from some directories to the local git repository folder and made some changes to the files. However, <code>git status</code> does not tell me the changes.<br>
It keeps saying, On branch master, Your branch is up to date with 'origin/master.</p>
<p>Thank you for your help.<br>
Best,<br>
Amare</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="adesalegn" data-post="6" data-topic="129228">
<div class="title">
<div class="quote-controls"></div>
 adesalegn:</div>
<blockquote>
<p>On branch master, Your branch is up to date with 'origin/master.</p>
</blockquote>
</aside>
<p>As long as you haven’t made any new commits that message is correct, unversioned/changed files that haven’t been committed yet should be listed below it.</p>
<p>If unversioned files don’t show up, check if they might be hidden by a <code>.gitignore</code> file.</p>

**Comment by Muser0030** [comment]
Kindly asist am having a simmilar issue . someone advice on what to do and the procedure to follow kindly. Thanks in advance

**Comment by fariba61** [comment]
Hi all
I want to push an existing project (written in VS Code) to a repository in GitHub
I typed the bellow commands in terminal (in VS Code)
git init
git add .
git commit -m "typed my commit message"
git remote add origin https://github.com/my-username/my-repository-name.git
git push -u -f origin master

but it didn't push and I had the bellow message
branch 'master' set up to track 'origin/master'.

**Reply by jacey10** [reply]
> Hi all I want to push an existing project (written in VS Code) to a repository in GitHub I typed the bellow commands in terminal (in VS Code) git init git add . git commit -m "typed my commit message" git remote add origin https://github.com/my-username/my-repository-name.git git push -u -f origin master
> 
> but it didn't push and I had the bellow message branch 'master' set up to track 'origin/master'.

You should use the command below:

git push -u origin master

That should work.

**Comment by Ramala27** [comment]
On branch main
Your branch is up to date with 'origin/main'.
